lib/internal/process/pre_execution.js

```diff
@@ -438,6 +440,54 @@ function setupInspectorHooks() {
   }
 }
 
+function setupNetworkInspection() {
+  if (!internalBinding('config').hasInspector) {
+    return;
+  }
+  const dc = require('diagnostics_channel');
+  const {
+    dataReceived,
+    loadingFinished,
+    requestWillBeSent,
+    responseReceived,
+  } = internalBinding('inspector');
+
+  let requestId = 0;
+  const getNextRequestId = () => `node-network-event-${++requestId}`;
+
+  dc.subscribe('http.client.request.start', ({ request }) => {
+    const url = `${request.protocol}//${request.host}${request.path}`;
+    const wallTime = DateNow();
+    const timestamp = wallTime / 1000;
+    request._inspectorRequestId = getNextRequestId();
+    requestWillBeSent(request._inspectorRequestId, url, request.method, timestamp, wallTime);
+  });
+  dc.subscribe('http.client.response.finish', ({ request, response }) => {
+    if (typeof request._inspectorRequestId !== 'string') {
+      return;
+    }
+    responseReceived(request._inspectorRequestId, DateNow() / 1000);
+    let responseString = '';
+    const onData = (chunk) => {
+      dataReceived(request._inspectorRequestId, DateNow() / 1000, chunk.length);
+      responseString += chunk.toString();
+    };
+    response.on('data', onData);
```

Comment:
Based on the test results, consuming response data and entering flowing mode in the diagnostics_channel hook is causing some issues in the core. Any ideas?

Reply:
Show the issues?

Reply:
https://github.com/nodejs/node/actions/runs/9696727280/job/26759268382?pr=53593
I'll investigate and summarize the issues later.

Reply:
Yeah this seems wrong - we can't consume the request this way :D

We'd need to 'tee' the stream (discussions happening elsewhere) though with the inspector cloning it should also be fine.

I think for this PR it would make sense to exclude request bodies and do it in a follow up PR.

cc @MoLow @mcollina (since our tee discussion elsewhere)

Reply:
Dropped support for response body inspection for now. fbf5a31af46d564c7530d70e93b672cdd22c448f